packages/rocketchat-lib/server/methods/leaveRoom.js

```diff
@@ -27,7 +27,15 @@ Meteor.methods({
 				throw new Meteor.Error('error-you-are-last-owner', 'You are the last owner. Please set new owner before leaving the room.', { method: 'leaveRoom' });
 			}
 		}
+		if (room.automatic) {
```

Comment:
Why not use the callback https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-lib/server/functions/removeUserFromRoom.js#L20 ?